### ISSUE TITLE
[Bug]: "404 Not Found" PsychoJS Library Contents

### ISSUE THREAD

**Issue by JartanFTW** [psychopy/psychopy#6862]
### PsychoPy Version

2024.2.1post4

### What OS are your PsychoPy running on?

Windows 11

### Bug Description

When piloting an experiment in-browser set to use version 2024.2.1post4 on the 2024.2.1post4 PsychoPy Builder, the files downloaded to `lib` each contain the same 404 Not Found content, and the browser experiment does not appear to function.

### Expected Behaviour

PsychoPy should download the appropriate PsychoJS library files.

### Steps to Reproduce

1. Create a new experiment.
2. Save the experiment somewhere.
3. Pilot the saved experiment in browser.
4. See contents of generated `lib` folder.
...

### Additional context

File contents:
```<html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

I reproduced this issue on multiple PsychoPy Builder versions.
There also may be an issue where the selected "Use PsychoPy version" experiment setting does not download the equivalent version PsychoJS library files. When trying older PsychoPy Builder versions, as well as well selecting an older "Use PsychoPy version" experiment setting, the application kept downloading the 2024.2.1post4 PsychoJS files.

**Comment by peircej** [comment]
Apologies for that - it's to do with the version number not being the normal format. It's fixed in 2024.2.2

**Comment by JartanFTW** [comment]
> Apologies for that - it's to do with the version number not being the normal format. It's fixed in 2024.2.2

Great, thank you! Has the cause of the issue (improper version number) been addressed to avoid the problem in the future? Either by a system to validate proper version numbers, or a system to dynamically adapt to improper version numbers, or something else?

**Comment by peircej** [comment]
The immediate fix was to release a version without that post tag. I have just [submitted a PR](https://github.com/psychopy/psychopy/pull/6873) that checks/strips it when generating the psychojs.

For full info, the post tag is a [valid Python version number following PEP 440](https://peps.python.org/pep-0440/#post-releases). The issue was just that our PsychoJS releases don't include those tags (the post4 release was just adding a missing file to that standalone installer which doesn't require a new release of psychojs). The PR above checks for this post tag in the same way it was previously checking for rc, dev, b, a so that 2025.1.0post10 will link correctly to JS v 2025.1.0
